### PR TITLE
Fix merge related bugs

### PR DESCRIPTION
**Pull request by chrimc62**
Improve code robustness around missing files.
Fix bug where file was copied from old.
Improved tests.

### CODE REVIEW

File: experimental/generation/generator/packages/library/src/mergeAssets.ts [hunk 1]
```diff
@@ -705,7 +703,7 @@ async function mergeDialogs(schemaName: string, oldPath: string, oldFileList: st
             insertTrigger(i, newTriggers, mergedTriggers)
             mergedTriggerMap.set(triggerName, trigger)
             if (typeof trigger === 'string') {
-                await copySingleFile(oldPath, mergedPath, triggerName + '.dialog', oldFileList, feedback)
+                await copySingleFile(newPath, mergedPath, triggerName + '.dialog', newFileList, feedback)
```
Comment (coldplaying42) [inline experimental/generation/generator/packages/library/src/mergeAssets.ts:706]: why this changes to new path?

Comment (chrimc62) [reply experimental/generation/generator/packages/library/src/mergeAssets.ts:706]: Because this over new triggers which weren't in old.
